src/Articulate/Controllers/MetaWeblogController.cs

```diff
@@ -0,0 +1,49 @@
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Articulate.MetaWeblog;
+using Examine;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Microsoft.Extensions.Logging;
+using Umbraco.Cms.Core.Web;
+using Umbraco.Cms.Web.Common.Controllers;
+using WilderMinds.MetaWeblog;
+
+namespace Articulate.Controllers
+{
+    public class MetaWeblogController : RenderController
+    {
+        private readonly ArticulateMetaWeblogProvider _articulateMetaWeblogProvider;
+        private readonly MetaWeblogService _metaWeblogService;
+
+        public MetaWeblogController(ILogger<RenderController> logger,
+            ICompositeViewEngine compositeViewEngine,
+            IUmbracoContextAccessor umbracoContextAccessor,
+            ArticulateMetaWeblogProvider articulateMetaWeblogProvider,
+            MetaWeblogService metaWeblogService)
+            : base(logger, compositeViewEngine, umbracoContextAccessor)
+        {
+            _articulateMetaWeblogProvider = articulateMetaWeblogProvider;
+            _metaWeblogService = metaWeblogService;
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> IndexAsync(int id)
+        {
+
+            var rawContent = string.Empty;
+            using(var reader = new StreamReader(Request.Body))
+            {
+                rawContent = reader.ReadToEnd();
+            }
+
+            // Set the Blog Root Node ID in the implementation
+            _articulateMetaWeblogProvider.ArticulateBlogRootNodeId = id;
+
+            // This service call from the Nuget package - consumes our implmentation/service of Metaweblog above
+            string result = await _metaWeblogService.InvokeAsync(rawContent);
```

Comment:
From what I can tell the articulate root node is always zero when your provider is executed because:

* You have registered both the concrete ArticulateMetaWeblogProvider and the abstraction of IMetaWeblogProvider as separate services
* Assuming that IMetaWeblogProvider is injected into this MetaWeblogService - which will be a different instance of the ArticulateMetaWeblogProvider instance you are modifying here

To fix that you need to change the DI registration to be a factory for IMetaWeblogProvider. So you would register the concrete ArticulateMetaWeblogProvider as you are doing, and then when registering the IMetaWeblogProvider you'd use `services.AddScoped<IMetaWeblogProvider>(s => s.GetRequiredService<ArticulateMetaWeblogProvider>());` which means that for both ArticulateMetaWeblogProvider and IMetaWeblogProvider you'd get the same instance.